src/Constraint/ConstraintCollectionBuilder.php

```diff
@@ -18,7 +18,7 @@ class ConstraintCollectionBuilder
      * @return Constraint|Constraint[]
      * @throws InvalidRuleException
      */
-    public function build(ConstraintMap $constraintsMap)
+    public function build(ConstraintMap $constraintsMap, bool $allowExtraFields = false)
```

Comment:
After some puzzling as the multitude of passing the `$allowExtraFields` bugged me a little. A different solution:

```
ConstraintFactory.php:57
$this->collectionBuilder->setAllowExtraFields(...)->build($constraintMap);
```

And just read `allowExtraFields` from the class property in ConstraintCollectionBuilder::109.

Reply:
Ye, wanted to discuss this with you further today. Added setter :)